site/app/controllers/student/SubmissionController.php

```diff
@@ -239,7 +239,14 @@ public function ajaxValidGradeable($gradeable_id) {
      * @Route("/{_semester}/{_course}/gradeable/{gradeable_id}/bulk", methods={"POST"})
     */
     public function ajaxBulkUpload($gradeable_id) {
-        $is_qr = $_POST['use_qr_codes'] === "true";
+        if (empty($_POST)) {
+            $max_size =  Utils::returnBytes(ini_get('post_max_size'));
+            return $this->uploadResult("Empty POST request. This may mean that the sum size of your files are greater than {$max_size}.", false);
+        }
+
+        if(isset($_POST['use_qr_codes'])){
+            $is_qr = $_POST['use_qr_codes'] === "true";
+        }
```

Comment:
If `!isset($_POST['use_qr_codes'])`, this causes `$is_qr` to be undefined for the check on line 351, which is a code smell. You should make sure all variables are defined when using them.

Also, this can be simplified into a one-liner:
`$is_qr = isset($_POST['use_qr_codes']) && $_POST['use_qr_codes'] === "true";`